src/strife/am_map.c

```diff
@@ -1235,7 +1235,12 @@ AM_drawFline_Vanilla
 	return;
     }
 
-#define PUTDOT(xx,yy,cc) fb[(yy)*f_w+(xx)]=(cc)
+#define PUTDOT_RAW(xx,yy,cc) fb[(yy)*f_w+(xx)]=(cc)
+#ifndef CRISPY_TRUECOLOR
+#define PUTDOT(xx,yy,cc) PUTDOT_RAW(xx,yy,cc)
+#else
+#define PUTDOT(xx,yy,cc) PUTDOT_RAW(xx,yy,(pal_color[(cc)]))
```

Comment:
This distinction is needed for `AM_drawFline_Smooth()`, right?

Reply:
Yes, absolutely. It is used in `AM_drawFline_Smooth`, just a carbon copy of implementation from [Doom](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/doom/am_map.c#L1308).